server/src/main/java/org/apache/uniffle/server/storage/HybridStorageManager.java

```diff
@@ -129,17 +121,12 @@ public void updateWriteMetrics(ShuffleDataFlushEvent event, long writeTime) {
 
   private StorageManager selectStorageManager(ShuffleDataFlushEvent event) {
     StorageManager storageManager = storageManagerSelector.select(event);
-    eventOfUnderStorageManagers.put(event, storageManager);
-    event.addCleanupCallback(() -> eventOfUnderStorageManagers.invalidate(event));
     return storageManager;
   }
 
   @Override
   public boolean write(Storage storage, ShuffleWriteHandler handler, ShuffleDataFlushEvent event) {
-    StorageManager underStorageManager = eventOfUnderStorageManagers.getIfPresent(event);
-    if (underStorageManager == null) {
-      return false;
-    }
+    StorageManager underStorageManager = selectStorageManager(event);
```

Comment:
`selectStorageManager` may return different `StorageManager` from the last calling this method because of full disks and corrupted disks. Is it ok?

Reply:
Partition data in different storage is normal for uniffle

Reply:
OK.